content/posts/event-driven-scaling-fission-function-using-keda.md

```diff
@@ -0,0 +1,239 @@
+2. As the message arrives in the event source - the Keda will scale the HPA and deployment from 0 - to 1 for consuming messages. As more messages arrive the deployment is scaled beyond 1 automatically too.
+3. The deployment is like an connector which consumes messages from the source and then calls a function. 
+4. The function consumes the message and puts the response in response topic and errors in error topic as may be applicable.
+
+# Tutorial
+
+The rest of the post will walk through a step by step guide for using the new message queue trigger.
+
+## Setup
+
+### Prerequisite
+- A Kubernetes cluster with the latest version of Fission
+- The latest version of Fission-CLI on the local machine
+
+### Setup KEDA
+
+We will be deploying KEDA using Helm 3. For more installation options, checkout [deploying KEDA](https://keda.sh/docs/1.5/deploy/).
+
+1. Add Helm repo
+```
+$ helm repo add kedacore https://kedacore.github.io/charts
+```
+2. Update Helm repo
+```
+$ helm repo update
+```
+3. Install KEDA Helm chart
+```
+$ kubectl create namespace keda
+$ helm install keda kedacore/keda --namespace keda
+```
+
+### Setup Apache Kafka
+
+There are various ways of using Kafka in the Kubernetes cluster. Here, we are going to use Strimzi, it simplifies the process of running Apache Kafka in a Kubernetes cluster. Follow the installation steps mentioned [here](https://strimzi.io/docs/operators/latest/quickstart.html#proc-install-product-str) and create a Kafka cluster.
+
+## Preparation for creating Message Queue Trigger
+
+Since we are done with the setup, now we will need following to create message queue trigger:
+
+1. Three Kafka topics
+2. Producer function
+3. Consumer function
+
+### Creating Kafka Topics
+
+Now, create three Kafka topics as mentioned below, replace namespace and cluster with yours.
+
+1. Kafka topic for function invocation
+```
+cat << EOF | kubectl create -n my-kafka-project -f -
+apiVersion: kafka.strimzi.io/v1beta1
+kind: KafkaTopic
+metadata:
+name: request-topic
+labels:
+    strimzi.io/cluster: "my-cluster"
+spec:
+partitions: 3
+replicas: 2
+EOF
+```
+2. Kafka topic for function response
+```
+cat << EOF | kubectl create -n my-kafka-project -f -
+apiVersion: kafka.strimzi.io/v1beta1
+kind: KafkaTopic
+metadata:
+name: response-topic
+labels:
+    strimzi.io/cluster: "my-cluster"
+spec:
+partitions: 3
+replicas: 2
+EOF
+```
+3. Kafka topic for error response
+```
+cat << EOF | kubectl create -n my-kafka-project -f -
+apiVersion: kafka.strimzi.io/v1beta1
+kind: KafkaTopic
+metadata:
+name: error-topic
+labels:
+    strimzi.io/cluster: "my-cluster"
+spec:
+partitions: 3
+replicas: 2
+EOF
+```
+
+### Creating Producer Function
+
+The producer function is a Go program that creates one thousand messages with the timestamp and drops into a topic input. For brevity, all values have been hardcoded in the code itself. Save the following code as producer.go
+
+```
```

Comment:
We will also need to add go.mod code here

Reply:
Done